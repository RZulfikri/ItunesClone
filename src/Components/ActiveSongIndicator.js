import React, {PureComponent} from 'react';
import Icons from 'react-native-vector-icons/MaterialCommunityIcons';
import {connect} from 'react-redux';
import {createSelector} from 'reselect';
import {ItunesSelectors} from '../Redux/ItunesRedux';
import Colors from '../Themes/Colors';

/**
 * Component to idicate active music/song in the list
 */

class ActiveSongIndicator extends PureComponent {
  render() {
    const {data, activeSong} = this.props;
    if (activeSong) {
      if (data.get('trackId') === activeSong.get('trackId')) {
        return <Icons name={'volume-high'} size={30} color={Colors.black} />;
      }
    }

    return null;
  }
}

const selector = createSelector([ItunesSelectors.getActiveSong], activeSong => {
  return {
    activeSong,
  };
});

const mapStateToProps = state => {
  return selector(state);
};

export default connect(mapStateToProps, null)(ActiveSongIndicator);
