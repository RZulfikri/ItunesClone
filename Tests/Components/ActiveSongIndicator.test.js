import 'react-native';
import React from 'react';
import ActiveSongIndicator from '../../src/Components/ActiveSongIndicator';
import renderer from 'react-test-renderer';
import {fromJS, Map} from 'immutable';
import configureStore from 'redux-mock-store';
import {Provider} from 'react-redux';
const mockStore = configureStore([]);

describe('ActiveSongIndicator React-Redux Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore(
      Map({
        itunes: fromJS({activeSong: {trackId: 1}}),
      }),
    );
  });

  it('should renders correctly if data.trackId equal to activeSong.trackId', () => {
    const data = Map({trackId: 1});
    const component = renderer.create(
      <Provider store={store}>
        <ActiveSongIndicator data={data} />
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('should not renders correctly if data.trackId not equal to activeSong.trackId', () => {
    const data = Map({trackId: 2});
    const component = renderer.create(
      <Provider store={store}>
        <ActiveSongIndicator data={data} />
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
