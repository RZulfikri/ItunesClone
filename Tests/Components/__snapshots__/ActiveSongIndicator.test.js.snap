// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveSongIndicator React-Redux Component should not renders correctly if data.trackId not equal to activeSong.trackId 1`] = `null`;

exports[`ActiveSongIndicator React-Redux Component should renders correctly if data.trackId equal to activeSong.trackId 1`] = `
<Text
  allowFontScaling={false}
  style={
    Array [
      Object {
        "color": "#000000",
        "fontSize": 30,
      },
      undefined,
      Object {
        "fontFamily": "Material Design Icons",
        "fontStyle": "normal",
        "fontWeight": "normal",
      },
      Object {},
    ]
  }
>
  ó°•¾
</Text>
`;
